packages/ember-application/lib/system/application.js

```diff
@@ -679,16 +561,29 @@ var Application = Namespace.extend({
     @private
     @method runInitializers
   */
-  runInitializers(registry) {
+  runInitializers() {
     var App = this;
     this._runInitializer('initializers', function(name, initializer) {
       Ember.assert("No application initializer named '" + name + "'", !!initializer);
 
       if (isEnabled("ember-application-initializer-context")) {
-        initializer.initialize(registry, App);
+        // TODO2.0 remove deprecated case in which initialize takes two args
+        if (initializer.initialize.length === 2) {
+          Ember.deprecate("The `initialize` method for Application initializer '" + name + "' should take only one argument - `App`, an instance of an `Application`.");
+          initializer.initialize(App.__registry__, App);
+        } else {
+          initializer.initialize(App);
+        }
       } else {
         var ref = initializer.initialize;
-        ref(registry, App);
+
+        // TODO2.0 remove deprecated case in which initialize takes two args
+        if (ref.length === 2) {
```

Comment:
Yes, that is the question. I'd hate to land this post-2.0 and carry forward all these deprecations until 3.0.